presto-spi/src/main/java/io/prestosql/spi/type/Timestamps.java

```diff
@@ -79,10 +79,6 @@ public static long round(long value, int magnitude)
      */
     static long rescale(long value, int fromPrecision, int toPrecision)
     {
-        if (value < 0) {
-            throw new IllegalArgumentException("value must be >= 0");
```

Comment:
`rescale` method works great when the rounding is known to be not necessary. 
For cases when rounding may be necessary, `round`, `floorDiv` seem to offer well defined semantics.

In this commit you're using the rounding-or-truncation of this method, but i'd change it to expect that value is pre-rounded. (actual check may be too expensive for runtime, idk).

This is more relevant when we allow negative numbers here.

Reply:
> rescale method works great when the rounding is known to be not necessary.

Correct, and that's the exact purpose of this method. The only reason we had this check is to protect against trying to rescale epoch values which can be negative. But it doesn't matter. I'll get rid of the call to rescale as you suggested above and revert this change.